cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/constants/AnomalyDetectorConfig.java

```diff
@@ -227,6 +228,31 @@ private AnomalyDetectorConfig() {
   public static final String DEFAULT_MAINTENANCE_EVENT_CLASS = MaintenanceEvent.class.getName();
   public static final String MAINTENANCE_EVENT_CLASS_DOC = "The name of class that extends maintenance event.";
 
+  /**
+   * <code>maintenance.event.enable.idempotence</code>
+   */
+  public static final String MAINTENANCE_EVENT_ENABLE_IDEMPOTENCE_CONFIG = "maintenance.event.enable.idempotence";
+  public static final boolean DEFAULT_MAINTENANCE_EVENT_ENABLE_IDEMPOTENCE = true;
+  public static final String MAINTENANCE_EVENT_ENABLE_IDEMPOTENCE_DOC = "The flag to indicate whether maintenance event detector will drop "
+      + "the duplicate maintenance events detected within the configured retention period.";
+
+  /**
+   * <code>maintenance.event.idempotence.retention.ms</code>
+   */
+  public static final String MAINTENANCE_EVENT_IDEMPOTENCE_RETENTION_MS_CONFIG = "maintenance.event.idempotence.retention.ms";
+  public static final long DEFAULT_MAINTENANCE_EVENT_IDEMPOTENCE_RETENTION_MS = TimeUnit.MINUTES.toMillis(3);
+  public static final String MAINTENANCE_EVENT_IDEMPOTENCE_RETENTION_MS_DOC = "The maximum time in ms to store events retrieved from the"
+      + " MaintenanceEventReader. Relevant only if idempotency is enabled (see " + MAINTENANCE_EVENT_ENABLE_IDEMPOTENCE_CONFIG + ").";
+
+  /**
+   * <code>maintenance.event.max.idempotence.cache.size</code>
+   */
+  public static final String MAINTENANCE_EVENT_MAX_IDEMPOTENCE_CACHE_SIZE_CONFIG = "maintenance.event.max.idempotence.cache.size";
+  public static final int DEFAULT_MAINTENANCE_EVENT_MAX_IDEMPOTENCE_CACHE_SIZE = 25;
```

Comment:
Even though 25 as default is probably good enough. But why not make the mac cache size larger (e.g. 1000)? I assume the cache entry is not too big in size.

Reply:
Maintenance events (e.g. planned maintenance events in Azure) are expected to be rare. In particular, a burst of up to `25` _distinct_ (i.e. not duplicate) maintenance events within the default retention period must be extremely rare. CC's anomaly detector manager cannot (and is not expected to) handle this many maintenance events in such a short period.

Presumably, pushing the limit to a higher value would be an exceptional case that would better handled based on user needs.